src/mesh/logical_location.hpp

```diff
@@ -0,0 +1,276 @@
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef MESH_LOGICAL_LOCATION_HPP_
+#define MESH_LOGICAL_LOCATION_HPP_
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <set>
+#include <vector>
+
+#include "utils/morton_number.hpp"
+
+namespace parthenon {
+
+struct RootGridInfo {
+  int level;
+  int nx1, nx2, nx3;
+  bool periodic1, periodic2, periodic3;
+  // Defaults to root grid of single block at the
+  // coarsest level
+  RootGridInfo()
+      : level(0), nx1(1), nx2(1), nx3(1), periodic1(false), periodic2(false),
+        periodic3(false) {}
+};
+
+//--------------------------------------------------------------------------------------
+//! \struct LogicalLocation
+//  \brief stores logical location and level of MeshBlock
+
+class LogicalLocation { // aggregate and POD type
+  // These values can exceed the range of std::int32_t even if the root grid has only a
+  // single MeshBlock if >30 levels of AMR are used, since the corresponding max index =
+  // 1*2^31 > INT_MAX = 2^31 -1 for most 32-bit signed integer type impelementations
+  std::int64_t lx1_, lx2_, lx3_;
+  MortonNumber morton_;
+  int level_;
+
+ public:
+  LogicalLocation(int lev, std::int64_t l1, std::int64_t l2, std::int64_t l3)
+      : lx1_(l1), lx2_(l2), lx3_(l3), level_(lev), morton_(lev, l1, l2, l3) {}
+  LogicalLocation() : LogicalLocation(0, 0, 0, 0) {}
+
+  const auto &lx1() const { return lx1_; }
+  const auto &lx2() const { return lx2_; }
+  const auto &lx3() const { return lx3_; }
+  const auto &level() const { return level_; }
+  const auto &morton() const { return morton_; }
+
+  // operators useful for sorting
+  bool operator==(LogicalLocation &ll) {
+    return ((ll.level() == level_) && (ll.lx1() == lx1_) && (ll.lx2() == lx2_) &&
+            (ll.lx3() == lx3_));
+  }
+  // LFR: These are old comparison operators. Greater gets used for sorting
+  //      the derefinenment list, but we may want to remove them at some
+  //      point to avoid confusion with the comparators below.
+  static bool Lesser(const LogicalLocation &left, const LogicalLocation &right) {
+    return left.level() < right.level();
+  }
+  static bool Greater(const LogicalLocation &left, const LogicalLocation &right) {
+    return left.level() > right.level();
+  }
```

Comment:
How about renaming them to be more specific in the meantime (assuming that removing them is a larger lift)?

Reply:
Ok, should have read my comment. Only `Greater` is used in one place in AMR. I just replace that use with a lambda and removed the old comparators.